### PR TITLE
METRON-203 General best practice and bug fixes

### PR DESCRIPTION
**Pull request by mmiklavc**
https://issues.apache.org/jira/browse/METRON-203

Fixed:
- Boolean.getBoolean() usage issues.
- Potential NPE problems
- equals/hashcode usage

### CODE REVIEW

**Comment by mmiklavc** [comment]
Tested on full-dev-vagrant - verified indexes are populating.

File: metron-platform/metron-data-management/src/main/java/org/apache/metron/dataloads/nonbulk/taxii/TaxiiHandler.java [hunk 1]
```diff
@@ -61,346 +63,334 @@
...
-            LOG.error(msg, cce);
-            throw new RuntimeException(msg, cce);
-        }
+  public static <RESPONSE_T, REQUEST_T> RESPONSE_T call( HttpClient taxiiClient
+      , URI endpoint
+      , REQUEST_T request
+      , HttpClientContext context
+      , Class<RESPONSE_T> responseClazz
+  ) throws JAXBException, IOException {
+    //TaxiiXml taxiiXml = xmlFactory.get().createTaxiiXml();
```
Comment (cestella) [inline metron-platform/metron-data-management/src/main/java/org/apache/metron/dataloads/nonbulk/taxii/TaxiiHandler.java]: Can we delete this commented code?

**Comment by cestella** [comment]
+1

**Comment by merrimanr** [comment]
+1

**Pull request closed without merging**